polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -395,12 +396,11 @@ async fn distribute_collation<Context>(
 		return Ok(())
 	}
 
-	// Determine which core the para collated-on is assigned to.
+	// Determine which core(s) the para collated-on is assigned to.
 	// If it is not scheduled then ignore the message.
-	let (our_core, num_cores) =
-		match determine_core(ctx.sender(), id, candidate_relay_parent, relay_parent_mode).await? {
-			Some(core) => core,
-			None => {
+	let (our_core, num_cores, elastic_scaling) =
```

Comment:
```suggestion
	let (our_cores, num_cores) =
```

Reply:
We can use `our_cores[0]` later in the code and `our_cores.len() > 1` to detect elastic scaling.